src/Entity/PageManager.php

```diff
@@ -71,50 +83,24 @@ public function createWithDefaults(array $defaults = []): PageInterface
         return $page;
     }
 
+    /**
+     * NEXT_MAJOR: keep only $this->fixPageUrl->fixUrl$page) in this method.
+     */
     public function fixUrl(PageInterface $page): void
     {
-        if ($page->isInternal()) {
-            $page->setUrl(null); // internal routes do not have any url ...
-
-            return;
-        }
-
-        // hybrid page cannot be altered
-        if (!$page->isHybrid()) {
-            $parent = $page->getParent();
-
-            if (null !== $parent) {
-                $slug = $page->getSlug();
-
-                if (null === $slug) {
-                    $slug = $this->slugify->slugify($page->getName() ?? '');
```

Comment:
instead of introducing `PageFixerInterface` and `PageFixerService` I would suggest to just keep this code here and adjust it so it works with either `SlugifyInterface` or Symfony's `SluggerInterface` :thinking:

Reply:
But it's possible to make SlugifyInterface works. 

I created an Adapter to make PageFixService work with both. But It was suggested to just pass Symfony SluggerInterface and ignore SlugifyInterface. 

If I pass `SlugifyInterface | SluggerInterface ` It will keep the code into repository. And it will stay there.

In case I want to refactor/extract this code,  I neeed to accept something like. 

`SlugifyInterface |SluggerInterface |PageFixInterface `, it will be more complex to refactor.

And IMO move this code to PageFixer isn't a big change at all.